File: 2025-09-11 - Lista Simples Encadeada/Exemplo_C#/ListaEncadeadaInterface.cs
// ------------------------------------------------------------------------
// Interface da Lista Simplesmente Encadeada
// Define os métodos que a implementação deve possuir
// ------------------------------------------------------------------------
public interface ListaEncadeadaInterface
{
   // Inserções
   void InserirNoInicio(int valor);
   void InserirNoFim(int valor);
   void InserirNaPosicao(int valor, int posicao);

   // Remoções
   void RemoverInicio();
   void RemoverPorValor(int valor);
   void RemoverNaPosicao(int posicao);

   // Buscas
   bool BuscarPorValor(int valor);
   int? ObterNaPosicao(int posicao);

   // Atualizações
   void AtualizarPorValor(int valorAntigo, int novoValor);
   void AtualizarNaPosicao(int posicao, int novoValor);

   // Impressão
   void ImprimirLista();
}

File: 2025-09-11 - Lista Simples Encadeada/Exemplo_Java/ListaEncadeadaInterface.java
// ------------------------------------------------------------------------
// Interface da Lista Simplesmente Encadeada
// Define os métodos que a implementação deve possuir
// ------------------------------------------------------------------------
public interface ListaEncadeadaInterface
{
   // ---------------------------------------------------------------------
   // Inserções
   void inserirNoInicio(int valor);
   void inserirNoFim(int valor);
   void inserirNaPosicao(int valor, int posicao);

   // ---------------------------------------------------------------------
   // Remoções
   void removerInicio();
   void removerPorValor(int valor);
   void removerNaPosicao(int posicao);

   // ---------------------------------------------------------------------
   // Buscas
   boolean buscarPorValor(int valor);
   Integer obterNaPosicao(int posicao);

   // ---------------------------------------------------------------------
   // Atualizações
   void atualizarPorValor(int valorAntigo, int novoValor);
   void atualizarNaPosicao(int posicao, int novoValor);

   // ---------------------------------------------------------------------
   // Impressão
   void imprimirLista();
}
